From: Cong Nguyen <congnguyenvan002@gmail.com>
Subject: [PATCH] rz/v2h: update LPDDR4 DDR parameters to vendor v3.0.3

Port the Renesas RZ/V2H LPDDR4 parameter set v3.0.3
(L4X.R2W32X16D8S32.ADED-0) onto the in-tree parameter table.

The vendor drop ships its arrays in the flat "dat0/dat1" layout
(param_phyinit_{1d,2d}_dat{0,1}). They are folded into the layout the
v2h ddr_setup_lpddr4.c driver already expects, so no driver change is
needed:

  param_phyinit_1d_dat0  -> phyinit_1d[]            (identical, 0 changes)
  param_phyinit_2d_dat0  -> phyinit_2d[]            (identical, 0 changes)
  param_phyinit_1d_dat1  -> param_phyinit_f_1d_0[]  (base 0x054000, 4 changes)
  param_phyinit_2d_dat1  -> param_phyinit_f_2d_0[]  (base 0x054000, 5 changes)

param_phyinit_c[], param_phyinit_i[] and param_phyinit_swizzle[] are
already byte-identical to the vendor drop and are left untouched. The
1D/2D PHY training images also match the vendor data exactly over their
whole length -- the in-tree arrays are zero-padded out to 0x4000 entries,
which the driver's fixed 0..0x3fff load loop requires -- so they are
unchanged as well.

param_setup_mc[] (!PLAT_DDR_ECC) is replaced by the vendor table
verbatim. Besides the 20 registers the vendor actually changed, this
reverts eight hand-tuned registers that fix_xhci_warning_access_mem.patch
had applied (0x0134, 0x0135, 0x0178, 0x017f, 0x0181, 0x02cd, 0x02cf,
0x02d0) in favour of the vendor values.

param_setup_mc[] (PLAT_DDR_ECC) receives the same vendor updates except
for the address-map registers 0x0130, 0x0132 and 0x0133, which are
legitimately ECC-specific and are kept at their existing values; the
vendor supplies a non-ECC table only. Note that v2h_common.mk sets
PLAT_DDR_ECC := 0, so this branch is not currently compiled.

LPDDR4_VERSION is bumped 1.06 -> 3.03 to match the ported data.

Upstream-Status: Pending

Signed-off-by: Cong Nguyen <congnguyenvan002@gmail.com>
---
--- a/plat/renesas/rz/soc/v2h/drivers/ddr/ddr_param_def_lpddr4.c
+++ b/plat/renesas/rz/soc/v2h/drivers/ddr/ddr_param_def_lpddr4.c
@@ -695,13 +695,13 @@
 	{0x054019, 0x2d54},
 	{0x05401a, 0x00b1},
 	{0x05401b, 0x4134},
-	{0x05401c, 0x1908},
+	{0x05401c, 0x1928},
 	{0x05401d, 0x0000},
 	{0x05401e, 0x0006},
 	{0x05401f, 0x2d54},
 	{0x054020, 0x00b1},
 	{0x054021, 0x4134},
-	{0x054022, 0x1908},
+	{0x054022, 0x1928},
 	{0x054023, 0x0000},
 	{0x054024, 0x002e},
 	{0x054025, 0x0000},
@@ -720,13 +720,13 @@
 	{0x054032, 0x5400},
 	{0x054033, 0xb12d},
 	{0x054034, 0x3400},
-	{0x054035, 0x0841},
+	{0x054035, 0x2841},
 	{0x054036, 0x0019},
 	{0x054037, 0x0600},
 	{0x054038, 0x5400},
 	{0x054039, 0xb12d},
 	{0x05403a, 0x3400},
-	{0x05403b, 0x0841},
+	{0x05403b, 0x2841},
 	{0x05403c, 0x0019},
 	{0x05403d, 0x2e00},
 	{0x05403e, 0x0000},
@@ -1517,7 +1517,7 @@
 	{0x05400c, 0x0000},
 	{0x05400d, 0x0000},
 	{0x05400e, 0x0000},
-	{0x05400f, 0x0f00},
+	{0x05400f, 0x0100},
 	{0x054010, 0x2080},
 	{0x054011, 0x0000},
 	{0x054012, 0x0310},
@@ -1530,13 +1530,13 @@
 	{0x054019, 0x2d54},
 	{0x05401a, 0x00b1},
 	{0x05401b, 0x4134},
-	{0x05401c, 0x1908},
+	{0x05401c, 0x1928},
 	{0x05401d, 0x0000},
 	{0x05401e, 0x0006},
 	{0x05401f, 0x2d54},
 	{0x054020, 0x00b1},
 	{0x054021, 0x4134},
-	{0x054022, 0x1908},
+	{0x054022, 0x1928},
 	{0x054023, 0x0000},
 	{0x054024, 0x002e},
 	{0x054025, 0x0000},
@@ -1555,13 +1555,13 @@
 	{0x054032, 0x5400},
 	{0x054033, 0xb12d},
 	{0x054034, 0x3400},
-	{0x054035, 0x0841},
+	{0x054035, 0x2841},
 	{0x054036, 0x0019},
 	{0x054037, 0x0600},
 	{0x054038, 0x5400},
 	{0x054039, 0xb12d},
 	{0x05403a, 0x3400},
-	{0x05403b, 0x0841},
+	{0x05403b, 0x2841},
 	{0x05403c, 0x0019},
 	{0x05403d, 0x2e00},
 	{0x05403e, 0x0000},
@@ -35099,24 +35099,24 @@
 	{0x0040, 0x11220408},
 	{0x0041, 0x04112204},
 	{0x0042, 0x01000000},
-	{0x0043, 0x01010001},
+	{0x0043, 0x01010101},
 	{0x0044, 0x00000003},
 	{0x0045, 0x00040301},
-	{0x0046, 0x00000260},
+	{0x0046, 0x000001c0},
 	{0x0047, 0x00001863},
-	{0x0048, 0x00000130},
+	{0x0048, 0x000000e0},
 	{0x0049, 0x00000c2e},
-	{0x004a, 0x00000026},
+	{0x004a, 0x0000001c},
 	{0x004b, 0x0000017f},
 	{0x004c, 0x00000005},
-	{0x004d, 0x00100130},
-	{0x004e, 0x00980090},
+	{0x004d, 0x001000e0},
+	{0x004e, 0x00700090},
 	{0x004f, 0x00004808},
-	{0x0050, 0x00040013},
-	{0x0051, 0x01300009},
+	{0x0050, 0x0004000e},
+	{0x0051, 0x00e00009},
 	{0x0052, 0x00009010},
-	{0x0053, 0x00080098},
-	{0x0054, 0x00130048},
+	{0x0053, 0x00080070},
+	{0x0054, 0x000e0048},
 	{0x0055, 0x00000904},
 	{0x0056, 0x11010000},
 	{0x0057, 0x0006000c},
@@ -35126,9 +35126,9 @@
 	{0x005b, 0x0a050503},
 	{0x005c, 0x02010303},
 	{0x005d, 0x00010101},
-	{0x005e, 0x026c026c},
-	{0x005f, 0x01360136},
-	{0x0060, 0x00270027},
+	{0x005e, 0x01cc01cc},
+	{0x005f, 0x00e600e6},
+	{0x0060, 0x001d001d},
 	{0x0061, 0x08030318},
 	{0x0062, 0x0c0c080c},
 	{0x0063, 0x06050203},
@@ -35242,7 +35242,7 @@
 	{0x00cf, 0x34340000},
 	{0x00d0, 0x34343434},
 	{0x00d1, 0x41414141},
-	{0x00d2, 0x08084141},
+	{0x00d2, 0x28284141},
 	{0x00d3, 0x19191919},
 	{0x00d4, 0x00001919},
 	{0x00d5, 0x00000000},
@@ -35335,9 +35335,9 @@
 	{0x012c, 0x00050028},
 	{0x012d, 0x01010100},
 	{0x012e, 0x00000202},
-	{0x012f, 0x00000000},
+	{0x012f, 0x00000101},
 	{0x0130, 0x6fff0000},
-	{0x0131, 0x000fff00},
+	{0x0131, 0x0007ff00},
 	{0x0132, 0xdfff7000},
 	{0x0133, 0x000fff01},
 	{0x0134, 0xff0b0000},
@@ -35415,9 +35415,9 @@
 	{0x017c, 0x00000008},
 	{0x017d, 0x02030306},
 	{0x017e, 0x01000101},
-	{0x017f, 0x01000202},
+	{0x017f, 0x01000304},
 	{0x0180, 0x01000304},
-	{0x0181, 0x01000505},
+	{0x0181, 0x01000304},
 	{0x0182, 0x01000304},
 	{0x0183, 0x00000304},
 	{0x0184, 0x00000000},
@@ -35749,10 +35749,10 @@
 	{0x02ca, 0xffffffff},
 	{0x02cb, 0x000f000f},
 	{0x02cc, 0x01000000},
-	{0x02cd, 0x01010401},
+	{0x02cd, 0x01010101},
 	{0x02ce, 0x00010101},
-	{0x02cf, 0x0301000a},
-	{0x02d0, 0x01010102},
+	{0x02cf, 0x01010064},
+	{0x02d0, 0x01010101},
 	{0x02d1, 0x00010101},
 	{0x02d2, 0x01010064},
 	{0x02d3, 0x01010101},
@@ -35793,7 +35793,7 @@
 	{0x02f6, 0x00200302},
 	{0x02f7, 0x00000000},
 	{0x02f8, 0x04001800},
-	{0x02f9, 0x01001804},
+	{0x02f9, 0x01011804},
 	{0x02fa, 0x01000101},
 	{0x02fb, 0x01000101},
 	{0x02fc, 0x01000100},
@@ -35878,24 +35878,24 @@
 	{0x0040, 0x11220408},
 	{0x0041, 0x04112204},
 	{0x0042, 0x01000000},
-	{0x0043, 0x01010001},
+	{0x0043, 0x01010101},
 	{0x0044, 0x00000003},
 	{0x0045, 0x00040301},
-	{0x0046, 0x00000260},
+	{0x0046, 0x000001c0},
 	{0x0047, 0x00001863},
-	{0x0048, 0x00000130},
+	{0x0048, 0x000000e0},
 	{0x0049, 0x00000c2e},
-	{0x004a, 0x00000026},
+	{0x004a, 0x0000001c},
 	{0x004b, 0x0000017f},
 	{0x004c, 0x00000005},
-	{0x004d, 0x00100130},
-	{0x004e, 0x00980090},
+	{0x004d, 0x001000e0},
+	{0x004e, 0x00700090},
 	{0x004f, 0x00004808},
-	{0x0050, 0x00040013},
-	{0x0051, 0x01300009},
+	{0x0050, 0x0004000e},
+	{0x0051, 0x00e00009},
 	{0x0052, 0x00009010},
-	{0x0053, 0x00080098},
-	{0x0054, 0x00130048},
+	{0x0053, 0x00080070},
+	{0x0054, 0x000e0048},
 	{0x0055, 0x00000904},
 	{0x0056, 0x11010000},
 	{0x0057, 0x0006000c},
@@ -35905,9 +35905,9 @@
 	{0x005b, 0x0a050503},
 	{0x005c, 0x02010303},
 	{0x005d, 0x00010101},
-	{0x005e, 0x026c026c},
-	{0x005f, 0x01360136},
-	{0x0060, 0x00270027},
+	{0x005e, 0x01cc01cc},
+	{0x005f, 0x00e600e6},
+	{0x0060, 0x001d001d},
 	{0x0061, 0x08030318},
 	{0x0062, 0x0c0c080c},
 	{0x0063, 0x06050203},
@@ -36021,7 +36021,7 @@
 	{0x00cf, 0x34340000},
 	{0x00d0, 0x34343434},
 	{0x00d1, 0x41414141},
-	{0x00d2, 0x08084141},
+	{0x00d2, 0x28284141},
 	{0x00d3, 0x19191919},
 	{0x00d4, 0x00001919},
 	{0x00d5, 0x00000000},
@@ -36114,13 +36114,13 @@
 	{0x012c, 0x00050028},
 	{0x012d, 0x01010100},
 	{0x012e, 0x00000202},
-	{0x012f, 0x00000000},
-	{0x0130, 0x7fff0000},
-	{0x0131, 0x000fff00},
-	{0x0132, 0xffff8000},
-	{0x0133, 0x000fff00},
-	{0x0134, 0x1f0b0000},
-	{0x0135, 0x0101011f},
+	{0x012f, 0x00000101},
+	{0x0130, 0x3fff0000},
+	{0x0131, 0x0007ff00},
+	{0x0132, 0x7fff4000},
+	{0x0133, 0x0007ff00},
+	{0x0134, 0xff0b0000},
+	{0x0135, 0x010001ff},
 	{0x0136, 0x01010101},
 	{0x0137, 0x1b010101},
 	{0x0138, 0x03000001},
@@ -36187,16 +36187,16 @@
 	{0x0175, 0x03050800},
 	{0x0176, 0x02020101},
 	{0x0177, 0x0c050c18},
-	{0x0178, 0x0a000808},
+	{0x0178, 0x0a020808},
 	{0x0179, 0x0b010405},
 	{0x017a, 0x08011000},
 	{0x017b, 0x0b080e00},
 	{0x017c, 0x00000008},
 	{0x017d, 0x02030306},
 	{0x017e, 0x00000101},
-	{0x017f, 0x00000202},
+	{0x017f, 0x00000404},
 	{0x0180, 0x00000404},
-	{0x0181, 0x01000505},
+	{0x0181, 0x00000404},
 	{0x0182, 0x00000404},
 	{0x0183, 0x00000404},
 	{0x0184, 0x00000000},
@@ -36528,10 +36528,10 @@
 	{0x02ca, 0xffffffff},
 	{0x02cb, 0x000f000f},
 	{0x02cc, 0x01000000},
-	{0x02cd, 0x01010401},
+	{0x02cd, 0x01010101},
 	{0x02ce, 0x00010101},
-	{0x02cf, 0x0301000a},
-	{0x02d0, 0x01010102},
+	{0x02cf, 0x01010064},
+	{0x02d0, 0x01010101},
 	{0x02d1, 0x00010101},
 	{0x02d2, 0x01010064},
 	{0x02d3, 0x01010101},
@@ -36572,7 +36572,7 @@
 	{0x02f6, 0x00200302},
 	{0x02f7, 0x00000000},
 	{0x02f8, 0x04001800},
-	{0x02f9, 0x01001804},
+	{0x02f9, 0x01011804},
 	{0x02fa, 0x01000101},
 	{0x02fb, 0x01000101},
 	{0x02fc, 0x01000100},
--- a/plat/renesas/rz/soc/v2h/drivers/ddr/ddr_private.h
+++ b/plat/renesas/rz/soc/v2h/drivers/ddr/ddr_private.h
@@ -12,7 +12,7 @@
 #include "ddr_regs.h"
 
 #define DDR4_VERSION	"1.06"
-#define LPDDR4_VERSION	"1.06"
+#define LPDDR4_VERSION	"3.03"
 
 #if !LPDDR4
 #define DDR_VERSION		DDR4_VERSION
